### PR TITLE
fix(postgres): reconnect on OperationalError too

### PR DESCRIPTION
**Pull request by ganwell**

### CODE REVIEW

**Comment by anehx** [comment]
@winged could you take a look at this as well?

File: manabi/lock.py [hunk 1]
```diff
@@ -243,6 +243,10 @@ def __init__(self, refresh: float, postgres_dsn: str):
         self._dict = ManabiPostgresDict(self, self._lock)
 
     def connect(self):
+        try:
+            self._connection.close()
```
Comment (winged) [inline manabi/lock.py:247]: Might need a comment here to un-confuse future readers. Specifically, that the `AttributeError` from not having `self._connection` at all on first call is also "ok" here
```suggestion
            # Try to close existing connection if there is one. 
            # On first call, this causes a AttributeError, which we can ignore as well
            self._connection.close()
```

**Review by winged — CHANGES_REQUESTED** [review]
Code looks good, I just think that we should drop some hints for future readers / admins regarding what's going on

**Comment by ganwell** [comment]
@winged I should have addressed everything.

**Review by winged — APPROVED** [review]

**Pull request merged**